src/Database/Query.php

```diff
@@ -149,11 +149,16 @@ class Query implements ExpressionInterface, IteratorAggregate, Stringable
     protected array $_resultDecorators = [];
 
     /**
-     * Statement object resulting from executing this query.
+     * @var \Cake\Database\Statement|null
+     */
+    protected ?Statement $_statement = null;
+
+    /**
+     * Result set from exeuted SELCT query.
      *
-     * @var \Cake\Database\StatementInterface|null
+     * @var iterable|null
      */
-    protected ?StatementInterface $_iterator = null;
+    protected ?iterable $_results = null;
```

Comment:
If we're adding new properties could we skip the `_` prefix?

Reply:
This is caused by `QueryTrait`. ORM\Query doesn't actually implement its own features. It uses `QueryTrait` which defines `$_results`. 

It's awkward to have a trait overrides the base class, but it seemed better to line it up.